asterisk_modules/app_grpcsttbackground/grpc_stt.cpp

```diff
@@ -89,6 +90,36 @@ static inline int aligned_samples(int samples)
 {
 	return (samples + ALIGNMENT_SAMPLES/2)/ALIGNMENT_SAMPLES*ALIGNMENT_SAMPLES;
 }
+static void log_send_time(bool * start_speaking, const std::string &x_request_id, std::size_t len)
+{
+    auto duration = std::chrono::system_clock::now();
+    long long seconds = std::chrono::duration_cast<std::chrono::seconds>(duration.time_since_epoch()).count();
+    long long millis = std::chrono::duration_cast<std::chrono::milliseconds>(duration.time_since_epoch()).count();
+    long t = std::chrono::system_clock::to_time_t(duration);
+    char buff[100];
+    if (!std::strftime(buff, 100, "%F %T", std::localtime(&t))) {
+        ast_log(AST_LOG_WARNING, "error getting current time");
+    }
+    if (*start_speaking) {
+        ast_log(AST_LOG_DEBUG, "send FIRST chunks to STT at %s.%lld x-request-id:%s data_len:%lu\n", buff, millis - seconds * 1000, x_request_id.c_str(), len);
+        *start_speaking = false;
```

Comment:
I don't know is it ok to perform side effects like this, but I think it would be better to make `log_send_time` more clear and do not change `start_speaking` here. Caller (on line 614) can change it itself